Mathlib/Data/Matroid/IndepAxioms.lean

```diff
@@ -93,10 +93,25 @@ variable {α : Type*}
 
 section IndepMatroid
 
-/-- A matroid as defined by the independence axioms. This is the same thing as a `Matroid`,
-  and so does not need its own API; it exists to make it easier to construct a matroid from its
-  independent sets. The constructed `IndepMatroid` can then be converted into a matroid
-  with `IndepMatroid.matroid`. -/
+/-- A matroid as defined by a ground set and an independence predicate.
+This definition is an implementation detail whose purpose is to organize the multiple
+different versions of the independence axioms;
+as a rule, the type `IndepMatroid` should never appear in type signatures outside this file.
+
+The intended pattern for defining a `Matroid α` from a known independence predicate
+`MyIndep : Set α → Prop` and ground set `E : Set α` is to write
+
+```
+def myMatroid : Matroid α := IndepMatroid.matroid <| IndepMatroid.ofFoo E MyIndep _ _ … _
+```
+
+where `IndepMatroid.ofFoo` is either `IndepMatroid.mk`,
+or one of the several other available constructors for `IndepMatroid`,
+and the `_` represent the proofs that this constructor requires.
```

Comment:
If I'm to follow this suggestion in #20887 without an intermediate IndepMatroid, I would have to write
```
  Matroid.copyBase (IndepMatroid.matroid
    { E := univ
      ... /- 34 more lines -/ })
    univ (fun s ↦ IsTranscendenceBasis R ((↑) : s → A)) rfl
    (fun B ↦ by simp_rw [Matroid.base_iff_maximal_indep, isTranscendenceBasis_iff_maximal]; rfl)
```
which is probably not very readable. Using `{ }` is less convenient than `where` but still nicer than using `IndepMatroid.mk` IMO. Even though you can still use `(Indep := _)`, `(indep_aug := _)` to indicate which field you are working on when using `mk`, you have to add the parentheses.